File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {

        int n = temperatures.size();
        vector<int> output(n,0);

        stack<int> st;

        for(int i = n-1 ; i>= 0;i--)
        {
            
            while(!st.empty() && temperatures[st.top()] <= temperatures[i])
            {
                    st.pop();
            }

            if(st.empty())
            {
                output[i] = 0;
                st.push(i);
            } 
            
            if(temperatures[st.top()] > temperatures[i])
            {
                output[i] = st.top() - i;
                
            }
            st.push(i);

            
                
            
        }

       

        return output;
    }
};

File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        Stack<int[]> st = new Stack<int[]>();
        int ans[] =new int[temperatures.length];
        
        for(int i = n - 1; i >= 0;i--)
        {
            while(!st.isEmpty() && st.peek()[0] <= temperatures[i])
            {
                st.pop();
            }
            if(!st.isEmpty())
            {
                ans[i] = st.peek()[1] - i;
                
            }
            st.push(new int[]{temperatures[i],i});
            
        }

        return ans;

    }
}